### PR TITLE
Add repository search

### PR DESCRIPTION
**Pull request by sloshy**
Was requested recently on Gitter, and I was able to stitch this together during some downtime. The repositories algebra now supports searching repositories globally.

### CODE REVIEW

File: github4s/src/main/scala/github4s/algebras/Repositories.scala [hunk 1]
```diff
@@ -69,6 +69,20 @@ trait Repositories[F[_]] {
...
+   *
+   * @param query The query string for the search.
+   * @param searchParams List of search params.
+   * @param headers Optional headers to include in the request.
+   * @return A GHResponse with the result of the search
+   */
+  def searchRepos(
+      query: String,
+      searchParams: List[SearchParam],
+      headers: Map[String, String] = Map.empty
```
Comment (kell18) [inline github4s/src/main/scala/github4s/algebras/Repositories.scala:83]: Hi! Thanks a lot for the PR!
Seems like [pagination](https://docs.github.com/en/rest/reference/search#search-repositories) argument is missing for this one... Or it's covered by SearchParams somehow?

Comment (sloshy) [reply github4s/src/main/scala/github4s/algebras/Repositories.scala:85]: Ah I just forgot about it. I've added it so it matches other parts of the algebras. Truthfully, I started with a copy of the issues search function which did not have a pagination parameter (maybe it should?) and that's the only reason I missed it.

**Review by kell18 — APPROVED** [review]
Looks good to me! 👍  But I'm not super acquainted with the project, so may miss some details :)

**Review by juanpedromoreno — COMMENTED** [review]
Thanks @sloshy , may I suggest adding some docs with the new feature?

**Comment by sloshy** [comment]
Thanks for the suggestion @juanpedromoreno I forgot about that. Docs are added that match the style of the previous docs :+1:

**Review by juanpedromoreno — APPROVED** [review]
:shipit:

**Review by BenFradet — DISMISSED** [review]
LGTM :+1: , do both issues and repos searches share all search params?

File: github4s/src/test/scala/github4s/integration/ReposSpec.scala [hunk 2]
```diff
@@ -452,4 +452,40 @@ trait ReposSpec extends BaseIntegrationSpec {
...
+
+    testIsRight[SearchReposResult](
+      response,
+      { r =>
+        r.total_count > 0 shouldBe true
+        r.items.nonEmpty shouldBe true
+      }
+    )
+    response.statusCode shouldBe okStatusCode
+  }
```
[the remark this answers is not in the document]
Comment (sloshy) [reply github4s/src/test/scala/github4s/integration/ReposSpec.scala:473]: Just pushed up a couple new search params, and changed the test to use them to search for sbt-microsites. Let me know if you had something a bit more complex in mind!

**Review by BenFradet — APPROVED** [review]

**Pull request merged**